src/ui/public/index_patterns/_index_pattern.js

```diff
@@ -222,32 +230,34 @@ define(function (require) {
         return body;
       };
 
-      function setId(id) {
-        return self.id = id;
-      }
-
       self.create = function () {
         var body = self.prepBody();
         return docSource.doCreate(body)
-        .then(setId)
         .catch(function (err) {
           if (_.get(err, 'origError.status') === 409) {
             var confirmMessage = 'Are you sure you want to overwrite this?';
 
             return safeConfirm(confirmMessage).then(
               function () {
-                return docSource.doIndex(body).then(setId);
+                return Promise.try(function () {
+                  const cached = patternCache.get(self.id);
```

Comment:
I think you should check for the existence of `self.id` before passing it to `patternCache.get()` or perhaps even throw an exception if someone even attempts to invoke `self.create()` before an id is set.